py/INN.py

```diff
@@ -155,9 +164,17 @@ def forward(
         blstm_out, _ = self.blstm(embedded_sentence)
         # unpacks the output tensor (re-adds the padding) so that other functions can use it
         blstm_out, _ = pad_packed_sequence(blstm_out)
+        '''
+        #taking the last layer of bert and switching batch size and sequence lenght to make it similar to blstm output
+        tokens = tokens.squeeze(0)
+        mask = mask.squeeze(0)
+
+        out = self.Bert(tokens,attention_mask=mask) 
+        bert_out = out[0]
+        bert_out = bert_out.permute(1,0,2)
         # gets the hidden vector for each entity and stores them in H
-        H = torch.randn(T.shape[0], curr_batch_size, 2 * HIDDEN_DIM).detach().to(self.device)
-        H = self.get_h_entities(entity_spans, blstm_out, H, curr_batch_size)
+        H = torch.randn(T.shape[0], curr_batch_size, self.word_embedding_dim).detach().to(self.device)       #changed 2 * HIDDEN_DIM to word_embedding_dim
+        H = self.get_h_entities(entity_spans, bert_out, H, curr_batch_size)
```

Comment:
We want to be able to run experiments with both models - one options would be to define separate classes INNModel and INNModelBERT, another option would be to add an embedding_method argument to the model that lets the user decide whether to use BERT, from-scratch embeddings, etc.